app/Http/Controllers/ProjectController.php

```diff
@@ -21,19 +24,58 @@ public function show($id) {
 	}
 
 	public function create() {
-		return view('projects.create');
+		return view(
+			'registerProject', [
+				'courses' => Auth::user()->EnrolledIn,
+				'categories' => Tag::where('type', 2)->get(),
+				'skillsets' => Tag::where('type', 1)->get()
+			]
+		);
 	}
 
 	public function store(Request $request) {
 		$attributes = request()->validate([
-			'name' => ['required'],
-			'video' => ['present'],
-			'long_description' => ['present'],
-			'short_description' => ['present'],
-			'photo' => ['present']
+			'projectName' => ['required'],
+			'videoPitch' => ['present'],
+			'longDescription' => ['present'],
+			'shortDescription' => ['present'],
+			'projectImage' => ['present'],
+			'category' => [
+				'present',
+				// verify category to be valid (exists as a category tag record)
+				Rule::in(Tag::where('type', 2)->pluck('id'))
+			],
+			'skillsets' => 'present|array|min:1',
+			// verify each elm in skillsets[] to exist as a skill tag record
+			'skillsets.*' => [
+				'integer',
+				Rule::in(Tag::where('type', 1)->pluck('id')),
+			]
 		]);
 
-		$project = Project::create($attributes);
-		return view('projects.show', ['project' => $project]);
+		$project = $this->saveRecord($attributes);
+		if (!$project->exists) {
+			abort(500);
+		}
+
+		if (isset($attributes['skillsets'])) {
```

Comment:
@hecerinc cuál es el mejor approach para validar si hay data aqui?

Reply:
https://github.com/ProyectoIntegrador2018/BlueMarket/pull/92#pullrequestreview-212631706

Reply:
Pues en teoría ya lo de arriba te lo validó, no? Is present, is array, min 1 length, exists in DB? O qué otra validación quieres agregar?

Reply:
true. thnks